tidy3d/components/simulation.py

```diff
@@ -1401,6 +1403,12 @@ def subsection(
         include_pml_cells : bool = False
             Keep PML cells in simulation boundaries. Note that retained PML cells will be converted
             to regular cells, and the simulation domain boundary will be moved accordingly.
+        skip_geometry_validation: bool = False
+            Skip geometry validation in generating new simulation object, since geometry is not modified
+            in this process. Other validators remain, and assuming that they at most interact with
+            the bounding box of the geometry.
+        deep_copy: bool = True
```

Comment:
this option is only used in the mode solver internals. but I guess is it possible we just always pass `deep=False` in the subsection method?

Basically, when would we want `deep=True` vs `deep=False` and can we just simplify it?

Reply:
While so far subsection is only used in mode solver internals, I feel it's a useful feature. So it's possible to be used elsewhere in the future. In that case, better to leave this option open?

Reply:
sure, but my question is rather: when we use it in the future, would we generally just use `deep=False`? or are there instances where we might want to deep copy?

Reply:
the same reason for default `deep=True` in `updated_copy`, if you want a subsection object that can be edited completely separate from the original simulation?

Reply:
ok sounds good then